dapr_agents/agents/agent/agent.py

```diff
@@ -0,0 +1,246 @@
+                if isinstance(response, ChatCompletion):
+                    response_message = response.get_message()
+                    if response_message:
+                        message_dict = {
+                            "role": "assistant",
+                            "content": response_message,
+                        }
+                        self.text_formatter.print_message(message_dict)
+
+                    if response.get_reason() == "tool_calls":
+                        tool_calls = response.get_tool_calls()
+                        if tool_calls:
+                            # Add the assistant message with tool calls to the conversation
+                            if response_message:
+                                # Extract content from response_message if it's a dict
+                                if isinstance(response_message, dict):
+                                    content = response_message.get("content", "")
+                                    if content is None:
+                                        content = ""
+                                    tool_calls_data = response_message.get(
+                                        "tool_calls", []
+                                    )
+                                else:
+                                    content = (
+                                        str(response_message)
+                                        if response_message is not None
+                                        else ""
+                                    )
+                                    tool_calls_data = []
+
+                                message_dict = {
+                                    "role": "assistant",
+                                    "content": content,
+                                    "tool_calls": tool_calls_data,
+                                }
+                                messages.append(message_dict)
+
+                            await self.process_response(tool_calls)
+                            for tool_msg in self.tool_history:
+                                tool_message_dict = {
+                                    "role": "tool",
+                                    "content": tool_msg.content or "",
+                                    "tool_call_id": tool_msg.tool_call_id,
+                                }
+                                messages.append(tool_message_dict)
+
+                            # Continue to next iteration to let LLM process tool results
+                            continue
+                    else:
+                        # Final response - add to memory and return
+                        content = response.get_content()
+                        if content:
+                            self.memory.add_message(AssistantMessage(content=content))
+                        # # TODO(@Sicoyle): we should not clear the tool history here, but rather when the agent is done, or not at all.
```

Comment:
we should always clear `self.tool_history` when the assistant returns a final response (i.e., when there are no more tool calls in the message and the conversation is ready to return a user-facing answer).

This ensures:
* Tool results are only kept for the current round of tool calls.
* We do not accidentally send tool results for tool calls that are no longer relevant in the next round.

Reply:
got it! Yeah, it would be good to give control to the user. With the open telemetry feature, all of that should be captured ;) , but for those not using the `observability` feature, it would be good to enable that capability.